File: backend/app/schemas/entry.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class EntryCreate(BaseModel):
    latitude: float | None = None
    longitude: float | None = None


class EntryResponse(BaseModel):
    id: UUID
    species_name_en: str
    species_name_pt: str
    description_en: str
    description_pt: str
    photo_url: str
    latitude: float | None = None
    longitude: float | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class EntryListResponse(BaseModel):
    entries: list[EntryResponse]
    total: int
    page: int
    per_page: int

File: frontend/src/types/entry.ts
export interface Entry {
  id: string;
  species_name_en: string;
  species_name_pt: string;
  description_en: string;
  description_pt: string;
  photo_url: string;
  latitude: number | null;
  longitude: number | null;
  created_at: string;
}

export interface EntryListResponse {
  entries: Entry[];
  total: number;
  page: number;
  per_page: number;
}
